setup

```diff
@@ -123,47 +126,75 @@ def copytree(src, dst, ignore_patterns=[]):
 
     _copytree()
 
+def run_command(command:List[str])->bool:
+    logger.debug(f"[Terminal] Running command `{command}`.")
+    try:
+        subprocess.check_output(command)
+    except subprocess.CalledProcessError as e:
+        logger.error(f"[Terminal] Running command {command} failed.")
+        return False
+    return True
 
-def build(args):
-    shutil.rmtree(MAGE_BUILD_DIRECTORY, ignore_errors=True)
-    os.makedirs(MAGE_BUILD_DIRECTORY, exist_ok=True)
-    copytree(LICENCE_DIRECTORY, MAGE_BUILD_DIRECTORY)
+def build_and_copy_cpp_modules(args:Dict[str,Any])->bool:
+    """
+    This function builds and copies C++ modules. Returns true if successfull, false otherwise.
+    """
 
-    #################################################
-    #              Load C++ modules                 #
-    #################################################
+    logger.info("[Terminal] (1/7) Building C++ modules started.")
 
     # Make build directory
     os.makedirs(CPP_BUILD_DIRECTORY, exist_ok=True)
 
-    # Start CMake building
+    # Prepare CMake flags
     os.chdir(CPP_BUILD_DIRECTORY)
-    subprocess.run(["ls", "-l"])
     cmake_args = [".."]
-    if args.gpu:
+    if args["gpu"]:
         cmake_args.append(f"-D{MAGE_GPU_BUILD}")
-    if args.cpp_build_flags is not None:
-        cmake_args += [f"-D{flag}" for flag in args.cpp_build_flags]
+    if args["cpp_build_flags"] is not None:
```

Comment:
extract args constants in enum or class with predefined values

Reply:
done